apps/antalmanac/src/components/RightPane/AddedCourses/AddedCoursePane.tsx

```diff
@@ -159,6 +160,112 @@ function CopyScheduleButton() {
     );
 }
 
+interface CustomEventsBoxProps {
+    customEvents: RepeatingCustomEvent[];
+    isSkeletonMode: boolean;
+    scheduleNames: string[];
+}
+
+function CustomEventsBox(props: CustomEventsBoxProps) {
+    const { scheduleNames, isSkeletonMode } = props;
+
+    const [customEvents, setCustomEvents] = useState(props.customEvents);
+
+    useEffect(() => {
+        const handleCustomEventsChange = () => {
+            setCustomEvents([...AppStore.schedule.getCurrentCustomEvents()]);
+        };
+
+        AppStore.on('customEventsChange', handleCustomEventsChange);
+        AppStore.on('currentScheduleIndexChange', handleCustomEventsChange);
+
+        return () => {
+            AppStore.off('customEventsChange', handleCustomEventsChange);
+            AppStore.off('currentScheduleIndexChange', handleCustomEventsChange);
+        };
+    }, []);
+
+    return (
+        <>
+            {customEvents.length > 0 && (
+                <Box>
+                    <Typography variant="h6">Custom Events</Typography>
+                    <Grid container spacing={1} padding={0}>
+                        {customEvents.map((customEvent) => {
+                            return (
+                                <Grid item md={12} xs={12} key={customEvent.title}>
+                                    <CustomEventDetailView
+                                        customEvent={customEvent}
+                                        scheduleNames={scheduleNames}
+                                        isSkeletonMode={isSkeletonMode}
+                                    />
+                                </Grid>
+                            );
+                        })}
+                    </Grid>
+                </Box>
+            )}
+        </>
```

Comment:
Because there's a top-level conditional, you can just have two different returns.

```suggestion
    if (customEvents.length <= 0) {
      return <Box />
    }
    
    return (
              <Box>
                  <Typography variant="h6">Custom Events</Typography>
                  <Grid container spacing={1} padding={0}>
                      {customEvents.map((customEvent) => {
                          return (
                              <Grid item md={12} xs={12} key={customEvent.title}>
                                  <CustomEventDetailView
                                      customEvent={customEvent}
                                      scheduleNames={scheduleNames}
                                      isSkeletonMode={isSkeletonMode}
                                  />
                              </Grid>
                          );
                      })}
                  </Grid>
              </Box>
```